time.c

```diff
@@ -2864,17 +2866,12 @@ obj2int(VALUE obj)
 static uint32_t
 obj2ubits(VALUE obj, size_t bits)
 {
-    static const uint32_t u32max = (uint32_t)-1;
-    const uint32_t usable_mask = ~(u32max << bits);
-    uint32_t rv;
-    int tmp = obj2int(obj);
+    const unsigned int usable_mask = (1 << bits) - 1;
```

Comment:
I think ((uint32_t)1 << bits) is better to avoid overflow to the sign bit.
Anyway, bits has precondition that 0 <= bits <= 31.
Some comment about that may help future developers.

Also, size_t may be uselessly big type for bits.

Reply:
Updated. I used `(1U << bits)` to make sure it's calculated unsigned, added precondition as a comment, and switched `bits` from `size_t` to `unsigned int`.